cache-invalidation/server.sh

```diff
@@ -0,0 +1,179 @@
+    # this.
+    echo 'TODO: this is BAD, needs a fix in the selinux-policy'
+    setenforce 0
+
+    if [ -n "${YUM_REPO}" ]
+    then
+	    yum-config-manager --add-repo=http://artifacts.ci.centos.org/nfs-ganesha/nightly/libntirpc/libntirpc-latest.repo
+	    yum-config-manager --add-repo=${YUM_REPO}
+
+	    # install the latest version of gluster
+	    yum -y install nfs-ganesha nfs-ganesha-gluster glusterfs-ganesha
+
+	    # start nfs-ganesha service
+	    if ! systemctl start nfs-ganesha
+	    then
+		    echo "+++ systemctl status nfs-ganesha.service +++"
+		    systemctl status nfs-ganesha.service
+		    echo "+++ journalctl -xe +++"
+		    journalctl -xe
+		    exit 1
+	    fi
+    else
+	    [ -n "${GERRIT_HOST}" ]
+	    [ -n "${GERRIT_PROJECT}" ]
+	    [ -n "${GERRIT_REFSPEC}" ]
+
+	    GIT_REPO=$(basename "${GERRIT_PROJECT}")
+	    GIT_URL="https://${GERRIT_HOST}/${GERRIT_PROJECT}"
+
+	    # install NFS-Ganesha build dependencies
+	    yum -y --enablerepo=centos-gluster*-test install glusterfs-api-devel
+	    yum -y install git bison flex cmake gcc-c++ libacl-devel krb5-devel \
+		    dbus-devel libnfsidmap-devel libwbclient-devel libcap-devel \
+		    libblkid-devel rpm-build redhat-rpm-config
+
+	    git init "${GIT_REPO}"
+	    pushd "${GIT_REPO}"
+
+	    git fetch "${GIT_URL}" "${GERRIT_REFSPEC}"
+	    git checkout -b "${GERRIT_REFSPEC}" FETCH_HEAD
+
+	    # update libntirpc
+	    git submodule update --init || git submodule sync
+
+	    mkdir build
+	    pushd build
+
+	    cmake -DCMAKE_BUILD_TYPE=Maintainer -DBUILD_CONFIG=everything ../src
+	    make dist
+	    rpmbuild -ta --define "_srcrpmdir $PWD" --define "_rpmdir $PWD" *.tar.gz
+	    rpm_arch=$(rpm -E '%{_arch}')
+	    ganesha_version=$(rpm -q --qf '%{VERSION}-%{RELEASE}' -p *.src.rpm)
+	    if [ -e ${rpm_arch}/libntirpc-devel*.rpm ]; then
+		    ntirpc_version=$(rpm -q --qf '%{VERSION}-%{RELEASE}' -p ${rpm_arch}/libntirpc-devel*.rpm)
+		    ntirpc_rpm=${rpm_arch}/libntirpc-${ntirpc_version}.${rpm_arch}.rpm
+	    fi
+	    yum -y install ${ntirpc_rpm} ${rpm_arch}/nfs-ganesha-{,gluster-}${ganesha_version}.${rpm_arch}.rpm
+
+	    # start nfs-ganesha service with an empty configuration
+	    > /etc/ganesha/ganesha.conf
+	    if ! systemctl start nfs-ganesha
+	    then
+		    echo "+++ systemctl status nfs-ganesha.service +++"
+		    systemctl status nfs-ganesha.service
+		    echo "+++ journalctl -xe +++"
+		    journalctl -xe
+		    exit 1
+	    fi
+    fi
+
+    # TODO: open only the ports needed?
+    # disable the firewall, otherwise the client can not connect
+    systemctl stop firewalld || service iptables stop
+
+    # install and start gluster
+    yum -y install glusterfs-server
+    systemctl start glusterd
+    mkdir -p /bricks/${GLUSTER_VOLUME}
+fi
+
+if [ $2 -eq 0 ]
+then
+    # create and start gluster volume
+    gluster peer probe $4
+    gluster volume create ${GLUSTER_VOLUME} \
+	replica 2 \
```

Comment:
just a note: replica-2 volumes are being deprecated. We may have to switch to arbiter/replica-3 for all gluster ci-tests in the near future.